src/json.hpp

```diff
@@ -1845,9 +1803,9 @@ class lexer
     @brief scan a string literal
 
     This function scans a string according to Sect. 7 of RFC 7159. While
-    scanning, bytes are escaped and copied into buffer yytext. Then the
-    function returns successfully, yytext is null-terminated and yylen
-    contains the number of bytes in the string.
+    scanning, bytes are escaped and copied into buffer yytext. Then the function
+    returns successfully, yytext is *not* null-terminated (as it may contain \0
+    bytes), and yytext.size() is the number of bytes in the string.
```

Comment:
Would it be more accurate to say that it's null terminated but may also contain embedded nulls?
This currently sounds like there isn't guaranteed to be at least one null.

Reply:
Ah, that's a change in behavior, I see now.

Reply:
Yes, I assume that it used to be NUL-terminated, in the mists of history, and was probably being returned as a `char*`'  Since the token can contain NULs (supplied by escaping), any assumptions about NUL and termination are incorrect.  So it was removed.